### DISCUSSION TITLE
alist+阿里云盘+rclone 加载问题

### DISCUSSION THREAD

**Discussion by Jue-Xu** [AlistGo/alist#2278] [Q&A]
目前，我用alist V3添加阿里云盘，然后rclone通过webdav挂载alist，问题是：
直接播放alist网页上（或者fileball添加webdav）阿里云盘的内容可以随意拖动进度条 非常流畅，但是rclone挂载alist到本地后，从本地文件夹播放则拖动进度条加载非常缓慢，比如 从电影中间开始播放好像要把前面的内容全部加载完才能播放. rclone 参数如下：
```
[Unit]
Description=alist RClone Mount Service
After=network-online.target

[Service]
Type=notify
ExecStart=/usr/bin/rclone mount \
  --config=/root/.config/rclone/rclone.conf \
  --log-level=INFO \
  --log-file=/opt/rclone/logs/rclone-mount.log \
  --user-agent=root \
  --header="Referer:" \
  --umask=002 \
  --gid=1002 \
  --uid=1000 \
  --allow-other \
  --timeout=1h \
  --poll-interval=15s \
  --dir-cache-time=1000h \
  --cache-dir=/opt/rclone/cache/alist \
  --vfs-cache-mode=full \
  --vfs-cache-max-size=1G \
  --vfs-cache-max-age=4h \
  alist: /root/alist
ExecStop=/bin/fusermount -uz /root/alist
Restart=on-abort
RestartSec=5
StartLimitInterval=60s
StartLimitBurst=3

[Install]
WantedBy=multi-user.target
```
不知道是rclone参数设置的问题，有什么办法debug？还是阿里云盘对alist和rclone的兼容问题，用相同的方法播放PikPak或者Google drive挂载到本地 都可以随意拖动进度条

**Comment by oneisall8955** [comment]
https://zsxwz.com/2022/08/12/windows使用alist生成各种网盘webdav，挂载到本地硬盘/

#比如挂载阿里云盘
rclone mount ali:/ W: --vfs-cache-mode writes --cache-dir E:\aliyun --network-mode --no-check-certificate --default-permissions --header "Referer:https://www.aliyundrive.com/" --vfs-read-chunk-size-limit 1G --vfs-read-chunk-size 64M --dir-cache-time 12h --buffer-size 32M

#--header "Referer:https://www.aliyundrive.com/" 挂载阿里云盘必须的参数。

**Reply by kokikoku** [reply]
52G片子，时长2小时20分钟。播放6分20秒左右卡住了。参数记录一下。

    --daemon \
    --use-mmap \
    --umask 000 \
    --allow-other \
    --allow-non-empty \
    --dir-cache-time 24h \
    --cache-dir /tmp/rclone \
    --vfs-cache-mode full \
    --buffer-size 512M \
    --vfs-read-chunk-size 16M \
    --vfs-read-chunk-size-limit 64M \
    --vfs-cache-max-size 10G \
    --config /root/.config/rclone/rclone.conf \
    --log-file /var/log/rclone.log \
    --log-level DEBUG \

**Comment by AKA-KC** [comment]
我用这个还挺快的 3秒左右加载出来
rclone mount Alist: /volume3/6T/Alist --use-mmap --umask 000 --allow-other --allow-non-empty --dir-cache-time 24h --cache-dir=/home/cache --vfs-cache-mode full --buffer-size 512M --vfs-read-chunk-size 16M --vfs-read-chunk-size-limit 64M --vfs-cache-max-size 10G --daemon
我是直接挂Alist的webdav的

**Reply by Jue-Xu** [reply]
我用了 这个参数 OK了。非常感谢！
好像是 --use-mmap 或者 --allow-non-empty 的作用

**Reply by Parbler** [reply]
> 我用了 这个参数 OK了。非常感谢！ 好像是 --use-mmap 或者 --allow-non-empty 的作用
大佬，/volume3/6T/Alist中的6T/Alist是目录，/home/cache是缓存所在的位置？

**Reply by Jue-Xu** [reply]
嗯，这个每个人设置的目录都不同

**Reply by sunlove01** [reply]
大佬你这个代码的自动挂载代码有吗？

**Reply by kokikoku** [reply]
> 我用这个还挺快的 3秒左右加载出来 rclone mount Alist: /volume3/6T/Alist --use-mmap --umask 000 --allow-other --allow-non-empty --dir-cache-time 24h --cache-dir=/home/cache --vfs-cache-mode full --buffer-size 512M --vfs-read-chunk-size 16M --vfs-read-chunk-size-limit 64M --vfs-cache-max-size 10G --daemon 我是直接挂Alist的webdav的

初入手rclone。播几分钟，就卡住了。回头试试。

**Reply by kokikoku** [reply]
真的可以了。哈哈。不过日志没了？
    --daemon \
    --use-mmap \
    --allow-non-empty \
    --vfs-cache-mode full \
    --vfs-cache-max-age 6h \
    --vfs-cache-max-size 10G \
    --buffer-size 64M \
    --dir-cache-time 5m \
    --poll-interval 1m \
    --vfs-read-ahead 128M \
    --network-mode \
    --tpslimit 10 \
    --tpslimit-burst 10 \
    --transfers 4 \
    --vfs-read-chunk-size 32M \
    --vfs-read-chunk-size-limit 512M \
    --cache-dir /tmp/rclone \
    --config /root/.config/rclone/rclone.conf \
    --log-file /var/log/rclone.log \
    --log-level DEBUG \

**Reply by kokikoku** [reply]
还是不行。感觉cache溢出。

**Reply by kokikoku** [reply]
这个是我暂时能用的参数。估计我内存没那么大。--vfs-cache-max-size 不能设10G。

    --daemon \
    --use-mmap \
    --allow-non-empty \
    --vfs-cache-mode full \
    --vfs-cache-max-age 1h \
    --vfs-cache-max-size 500M \
    --buffer-size 16M \
    --dir-cache-time 5m \
    --poll-interval 1m \
    --vfs-read-ahead 128M \
    --network-mode \
    --tpslimit 10 \
    --tpslimit-burst 10 \
    --transfers 4 \
    --vfs-read-chunk-size 5M \
    --vfs-read-chunk-size-limit 50M \
    --cache-dir /tmp/rclone \
    --config /root/.config/rclone/rclone.conf \
    --log-file /var/log/rclone.log \
    --log-level INFO \

**Comment by yoyo3287258** [comment]
我也是用rclone ，没问题

**Reply by lauren12133** [reply]
@Jue-Xu 你还在用吗 想用docker的  但是config文件不会写 最终的config挂载文件 能帮忙提供下吗

**Reply by Jue-Xu** [reply]
我目前用的这个config，alist主要挂载aliyun和Google drive，运行正常（之前aliyun的问题，用这个配置解决了）
```
[Unit]
Description=Alist RClone Mount Service
After=network-online.target

[Service]
Type=notify
ExecStart=/usr/bin/rclone mount \
  --config=/root/.config/rclone/rclone.conf \
  --log-level=INFO \
  --log-file=/opt/rclone/logs/rclone-mount.log \
  --user-agent=root \
  --header="Referer:" \
  --umask=002 \
  --gid=1002 \
  --uid=1000 \
  --allow-other \
  --allow-non-empty \
  --use-mmap \
  --timeout=1h \
  --poll-interval=15s \
  --dir-cache-time=5m \
  --cache-dir=/opt/rclone/cache/alist \
  --vfs-cache-mode=full \
  --vfs-cache-max-size=1G \
  --vfs-cache-max-age=4h \
  --vfs-read-chunk-size-limit=1G \
  --vfs-read-chunk-size=64M \
  --buffer-size=32M \
  alist: /root/alist
ExecStop=/bin/fusermount -uz /root/alist
Restart=on-abort
RestartSec=5
StartLimitInterval=60s
StartLimitBurst=3

[Install]
WantedBy=multi-user.target
```
注意修改你自己的各种路径

**Reply by lauren12133** [reply]
你也是docker吗  配置rclone 挂载那部分怎么操作 用不好这个项目 就是liux端rclone config 配置alist这步 在docker项目下怎么操作 我看你这个config文件只是个运行文件

**Reply by Jue-Xu** [reply]
我没有太明白你的情况。
我的rclone不是用docker安装的，上面的配置是Linux的system service文件 使得rclone以这些参数 开机自启动。另外，alist是docker安装。

**Reply by mrabit** [reply]
> @Jue-Xu 你还在用吗 想用docker的 但是config文件不会写 最终的config挂载文件 能帮忙提供下吗

我目前 docker-compose.yml 这样写的  你可以参考一下

```shell
  rclone_alist:
    container_name: rclone_alist
    image: rclone/rclone
    privileged: true
    volumes:
      - /root/.config/rclone:/config/rclone
      - /rclone/.cache:/.cache
      - /rclone/aliyun:/mount:shared
    environment:
      - TZ=Asia/Shanghai
    network_mode: "bridge"
    cap_add:
      - SYS_ADMIN
    security_opt:
      - apparmor:unconfined
    devices:
      - /dev/fuse
    command: "mount alist:/阿里云盘 /mount --cache-dir /.cache -v --copy-links --no-gzip-encoding --no-check-certificate --allow-other --allow-non-empty --buffer-size=256M --vfs-read-chunk-size=256M --vfs-read-chunk-size-limit=500M --vfs-cache-mode=full --vfs-cache-max-size=15G --dir-cache-time=1h --vfs-cache-max-age=96h --poll-interval=10m"
    external_links:
      - alist
```

**Reply by Jue-Xu** [reply]
学到了，原来可以把mount 命令放在 docker compose里

**Reply by xchydb** [reply]
@Jue-Xu 请教一下您的配置文件就是将rclone.sevice放入/etc/systemd/system目录，通过systemctl start命令运行后就能使用吗，我按照你的配置文件重复这个过程，但是jellyfin同样在拖动进度条时会把前面的内容全部加载完才能播放，我的配置文件如下：
[Unit]
Description=Alist RClone Mount Service
After=network-online.target

[Service]
Type=notify
ExecStart=/usr/bin/rclone mount \
  --config=/root/.config/rclone/rclone.conf \
  --log-level=INFO \
  --log-file=/opt/rclone/logs/rclone-mount.log \
  --user-agent=root \
  --header="Referer:" \
  --umask=002 \
  --gid=1002 \
  --uid=1000 \
  --allow-other \
  --allow-non-empty \
  --use-mmap \
  --timeout=1h \
  --poll-interval=15s \
  --dir-cache-time=5m \
  --cache-dir=/tmp \
  --vfs-cache-mode=full \
  --vfs-cache-max-size=1G \
  --vfs-cache-max-age=4h \
  --vfs-read-chunk-size-limit=1G \
  --vfs-read-chunk-size=64M \
  --buffer-size=32M \
  alist: /mnt/webdav/alist
ExecStop=/bin/fusermount -uz /mnt/webdav/alist
Restart=on-abort
RestartSec=5
StartLimitInterval=60s
StartLimitBurst=3

[Install]
WantedBy=multi-user.target
望解答，感谢

**Reply by Jue-Xu** [reply]
我用这个配置在jellyfin下也是能推动进度条的。（你用哪个版本的alist

**Reply by xchydb** [reply]
官方原版的v3.12.2

**Reply by xchydb** [reply]
日志文件里有这么一条：2023/03/26 00:20:59 INFO  : webdav root '': poll-interval is not supported by this remote
请教下有没有可能就是这儿出现了问题

**Reply by Jue-Xu** [reply]
> 官方原版的v3.12.2

alist最新版本是v3.15.0，更新一下试试

**Reply by Jue-Xu** [reply]
> 日志文件里有这么一条：2023/03/26 00:20:59 INFO : webdav root '': poll-interval is not supported by this remote 请教下有没有可能就是这儿出现了问题

不清楚这个参数的意义，也是从别人那里抄来的，我的经验是 `--use-mmap`  或者 `--allow-non-empty` 起了作用
